src/sys/windows/mod.rs

```diff
@@ -22,6 +22,20 @@ macro_rules! syscall {
     }};
 }
 
+/// Helper macro to execute an I/O operation and register interests if the
+/// operation would block.
+macro_rules! try_io {
+    ($self: ident, $method: ident $(, $args: expr)*)  => {{
+        let result = (&$self.inner).$method($($args),*);
+        if let Err(ref e) = result {
+            if e.kind() == io::ErrorKind::WouldBlock {
+                $self.io_blocked_reregister()?;
+            }
+        }
+        result
+    }};
+}
```

Comment:
I think it can also reregister on successful operation.

```
macro_rules! try_io {
    ($self: ident, $method: ident $(, $args: expr)*)  => {{
        let result = (&$self.inner).$method($($args),*);
        if let Err(ref e) = result {
            if e.kind() == io::ErrorKind::WouldBlock {
                $self.io_blocked_reregister()?;
            }
        } else {
            $self.io_blocked_reregister()?;
        }
        result
    }};
}
```

Reply:
Do you mean we have to register on successful operations? Because I've kept the meaning of the original code.

Reply:
Because the events are deregistered on event received. We should reregister on every successful operation.

Reply:
Maybe we want to always reregister then? I'm not sure if we get more events on kqueue/epoll after an I/O operation returned an error.

Reply:
Are there are cases in which we can't reregister?

Reply:
I don't think notification will come up if we failed to perform these operations, so I want to let the reregister happened on sucessful registeration.

Reply:
Done.

Reply:
There are some place to modify to pass the test suite (except try_clone on TcpListener and several minors):
mark delete the sockstate in deregister.
wait for writable event on ttl test in tcp_stream.rs

I will send pr after you done this

Reply:
I'm not quite sure. But accept will fail if you successfully accepted one connection and continue to listen for accept event

Reply:
I've dropped the commit from the rebased commits I've pushed to see if the CI is green.